GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -229,6 +234,27 @@ public void Initialize([CanBeNull]IAheadBehindDataProvider aheadBehindDataProvid
             _ = UICommandsSource;
         }
 
+        public void SelectionChanged(IReadOnlyList<GitRevision> selectedRevisions)
+        {
+            var cancellationToken = _selectionCancellationTokenSequence.Next();
+
+            string selectedGuid = selectedRevisions.FirstOrDefault()?.Guid;
+            ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                var mergedBranches
+                    = selectedGuid == null ? new HashSet<string>()
+                    : (await Module.GetMergedBranchesAsync(includeRemote: true, fullRefname: true, commit: selectedGuid)).ToHashSet();
```

Comment:
Note: This adds 100 ms CPU time for me in a GE repo with 600 ms on my local relative new PC with Windows Defender. As a comparision git-status requires 150 ms. On repos with viruses like Symantec this can be substantially more even if git-status should have a greater hit (as more files are read there). This is async and will not affect the interactivity and should have a minimal impact on the perceived CPU use by GE.